### ISSUE TITLE
Target profile page rewrite: Create "Pathways" summary widget

### ISSUE THREAD

**Issue by andrewhercules** [opentargets/platform#251]

**Comment by andrewhercules** [comment]
### Use case(s)

1. As a user, I want to know if a gene is involved in any relevant pathways or biological processes

### Summary widget (showing different states depending on data)

![pathways](https://user-images.githubusercontent.com/7490258/49019287-7e7f0c80-f185-11e8-9681-da549e33965f.png)

**[Full size version](https://drive.google.com/open?id=1E_jJuN-uONZwsIug1k9DiySw64rmjD-7)**
 
### Key design and interaction notes

- If we have data for a given gene, please make the widget icon, widget border, and widget text dark grey (#5A5F5F) 
- If we do not have data for a given gene, please make widget icon, widget border, and widget text light grey (#E2DFDF) 

### Design assets

1. The widget icon is available on the [OT Core Platform Team Google Drive](https://drive.google.com/drive/folders/1w0CZuIXhbs7qCM7-mM9txCLvfhrxzgiB)
2. Colour codes are available on the [OT Core Platform Team Google Drive](https://drive.google.com/open?id=1d0JDmCznInsANZoQC77gadTf9jO0UDOo)

**Comment by peatroot** [comment]
@andrewhercules, is it possible to try a design with the top level pathways shown as on/off? I think this would be more useful than the number of pathways, but it's quite a bit of text. What do you think? The 27 possible top level pathways from Reactome are listed below.
```
Cell Cycle
Cell-Cell communication
Cellular responses to external stimuli
Chromatin organization
Circadian Clock
Developmental Biology
Digestion and absorption
Disease
DNA Repair
DNA Replication
Extracellular matrix organization
Gene expression (Transcription)
Hemostasis
Immune System
Metabolism
Metabolism of proteins
Metabolism of RNA
Mitophagy
Muscle contraction
Neuronal System
Organelle biogenesis and maintenance
Programmed Cell Death
Protein localization
Reproduction
Signal Transduction
Transport of small molecules
Vesicle-mediated transport
```

**Comment by andrewhercules** [comment]
It's a great idea @peatroot, but unfortunately, it's too much text having all 27 listed. Even having a few listed could be problematic - one user suggested listing the most important pathway, but others said that could be difficult to calculate as some users may be more concerned a gene is in one pathway versus another.

Perhaps we could use this idea in the detail view?

**Comment by peatroot** [comment]
Sure, we can do that. I find the number in the design unhelpful, because I know neither what set of pathways is being referred to, nor the size of that set.

**Comment by AsierGonzalez** [comment]
Personally, I find the number of pathways more informative than having the list of top level pathways, since these can be too generic. As a user, I would expect the actual list of pathways (not the top level ones) to appear when you click on the widget. Does that make any sense?

**Comment by peatroot** [comment]
@AsierGonzalez, I don't know what the plan is for the detail view, but likely similar to what you describe (and we currently have). We currently show the following (for ESR1):

<img width="940" alt="screenshot 2018-11-27 at 11 25 54" src="https://user-images.githubusercontent.com/7376272/49078939-47693380-f237-11e8-8ea1-ce4fba702156.png">

In addition to the show information, the API provides the top level pathway that each lower level pathway falls under. In the case of ESR1, the 13 lower level pathways listed cover 3 (of 27) top level pathways. I thought it might be a useful overview to see the 3 highlighted, as below:

<img width="440" alt="screenshot 2018-11-19 at 13 13 27" src="https://user-images.githubusercontent.com/7376272/49079073-9b741800-f237-11e8-9c62-bacbf3b19ebb.png">

If we just display a number:
* Should the number refer to lower level pathways, top level pathways or both combined?
* Do all users use Reactome for pathways? My understanding was that there are alternatives, which makes the number harder for users to interpret.

**Comment by AsierGonzalez** [comment]
Thanks @peatroot, I find the current approach quite convenient. The problem I see with the 27 top levels pathways from Reactome is that I believe that they are Reactome specific so people that don't know Reactome in detail (e.g. me) may not find them straightforward to understand. On the other hand, pathways are more specific, and even if users don't know what they are in detail, the names usually are descriptive enough.

If you would like to give some context, perhaps we could state the total number of pathways (currently 2244) on the widget, e.g. "14 Biological Processed and pathways involving ESR1 (out of 2244)"? By the way, and this may for another issue, is it 14 like on the widget or 13 like on the platform? Perhaps the difference due to the platform using and older version of Reactome?

**Comment by andrewhercules** [comment]
@peatroot and @AsierGonzalez: all are great points and worth considering for future implementations. For the initial proof-of-concept deadline, we will just stick with showing the number of low-level pathways as a number.

@AsierGonzalez, the discrepancy is my fault - I used data from an older release to design the static mockups :)

**Comment by AsierGonzalez** [comment]
Thank you for the clarification @andrewhercules

**Comment by peatroot** [comment]
Proposed schema:
```
type TargetSummaryPathways {
  count: Int!
}
```

**Comment by andrewhercules** [comment]
The layout of this widget may need to be updated once the max-width container is implemented - see issue #366 for further details.
